nano/node/rocksdb/rocksdb.cpp

```diff
@@ -290,21 +290,25 @@ int nano::rocksdb_store::decrement (nano::write_transaction const & transaction_
 	auto status = get (transaction_a, table_a, key_a, value);
 	release_assert (success (status));
 	auto base = static_cast<uint64_t> (value);
-	return put (transaction_a, table_a, key_a, nano::rocksdb_val (base - amount_a));
+	return put (transaction_a, table_a, key_a, nano::rocksdb_val (base - amount_a), store_hint::none);
 }
 
-int nano::rocksdb_store::put (nano::write_transaction const & transaction_a, tables table_a, nano::rocksdb_val const & key_a, nano::rocksdb_val const & value_a)
+int nano::rocksdb_store::put (nano::write_transaction const & transaction_a, tables table_a, nano::rocksdb_val const & key_a, nano::rocksdb_val const & value_a, nano::store_hint store_hint_a)
 {
 	debug_assert (transaction_a.contains (table_a));
 
 	auto txn = tx (transaction_a);
 	if (is_caching_counts (table_a))
 	{
-		if (!exists (transaction_a, table_a, key_a))
+		if (store_hint_a == store_hint::key_not_exists || (store_hint_a == store_hint::none && !exists (transaction_a, table_a, key_a)))
 		{
 			// Adding a new entry so counts need adjusting
 			increment (transaction_a, tables::cached_counts, rocksdb_val (rocksdb::Slice (table_to_column_family (table_a)->GetName ())), 1);
 		}
+		else
+		{
+			debug_assert (store_hint_a == store_hint::key_exists && exists (transaction_a, table_a, key_a));
```

Comment:
If `store_hint::none` (no hint given) and key does exist it seems like this would assert